### PR TITLE
Essay proposal - The Rust language build pipelines

### PR DESCRIPTION
**Pull request by arxra**
In the beginning of the course Monperrus suggested to investigate the rust language build pipeline in an essay, a build so massive it must be generated with scripts to build all its targets based on advanced rules.
This essay would take the reader through that pipeline, where it has been and how its doing.

### CODE REVIEW

**Comment by arxra** [comment]
Not again... my bad @cesarsotovalero , I'll learn how to not include already merged commits any year now.

**Review by cesarsotovalero — COMMENTED** [review]
Hi @arxra, I don't see a clear connection between buildpipe and Rust. Can you give more details of what you plan to cover in this essay?

**Comment by arxra** [comment]
The Rust programming language itself is tested using a complex pipeline which is what we want to cover. The project is https://github.com/rust-lang/rust and the pipelines run at https://dev.azure.com/rust-lang/rust .

**Comment by cesarsotovalero** [comment]
> The Rust programming language itself is tested using a complex pipeline which is what we want to cover. The project is [rust-lang/rust](https://github.com/rust-lang/rust) and the pipelines run at [dev.azure.com/rust-lang/rust](https://dev.azure.com/rust-lang/rust) .

I understand. Please edit the proposal to make it more clear. Then I'll merge your PR.

File: contributions/essay/aronber-yannik/Readme.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# The rust language build pipeline
+
+In this essay we would like to take the readers through a deep dive into how complex a buildpipe can be.
+One of the larger open-source build pipelines is the rust programming language buildpipe,
+with lots of buildtargets, advanced conditions and bots for building just what is needed for different events.
```
Comment (cesarsotovalero) [inline contributions/essay/aronber-yannik/Readme.md:5]: Add more details, eg. outline the sections of your essay.

Comment (ysndr) [reply contributions/essay/aronber-yannik/Readme.md:5]:
```suggestion
In this essay we would like to take the readers through a deep dive into how complex a buildpipe can be.
One of the larger open-source build pipelines is the rust programming language buildpipe,
with lots of buildtargets, advanced conditions and bots for building just what is needed for different events.

## Preliminary Outline
-  introduction
  - What is Rust
  - What is a buildpipe
  - What are GitHub Actions
- Parts of the Rust Build buildpipe
  - bots
     - [\@bors](https://bors.rust-lang.org/)
     - [\@rustbot](https://github.com/rust-lang/triagebot)
     - docs.rs
   - actions
      - docs
      - trying builds
      - releasing
- General Contribution process (linking bots and actions together)
```

**Comment by arxra** [comment]
@ysndr merged your suggestions, @cesarsotovalero is this outline well defined enough?

**Comment by cesarsotovalero** [comment]
> @ysndr merged your suggestions, @cesarsotovalero is this outline well defined enough?

LGTM! good luck!

**Comment by Adamih** [comment]
@arxra  I will provide you with feedback when the Essay is submitted.

**Pull request merged**